File: frontend/src/app/models/contact-form.model.ts
export interface ContactFormModel {
  firstName:      string;
  lastName:       string;
  email:          string;
  phone:          string;
  subject:        string;
  message:        string;
  attachmentUrl?:  string | null;
  attachmentName?: string | null;
}

export interface UploadUrlResponse {
  uploadUrl: string;
  blobUrl:   string;
}

export interface ContactFormResponse {
  success:     boolean;
  message:     string;
  referenceId: string;
  submittedAt: string;
}

export interface ContactSubmission {
  id:             number;
  firstName:      string;
  lastName:       string;
  email:          string;
  phone:          string;
  subject:        string;
  message:        string;
  referenceId:    string;
  submittedAt:    string;
  attachmentUrl:  string | null;
  attachmentName: string | null;
}

File: backend/Models/ContactFormModel.cs
namespace ContactFormApi.Models;

public class ContactFormModel
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName  { get; set; } = string.Empty;
    public string Email     { get; set; } = string.Empty;
    public string Phone     { get; set; } = string.Empty;
    public string Subject   { get; set; } = string.Empty;
    public string Message        { get; set; } = string.Empty;
    public string? AttachmentUrl  { get; set; }  // blob URL after direct upload
    public string? AttachmentName { get; set; }  // original filename
}

public class ContactFormResponse
{
    public bool   Success   { get; set; }
    public string Message   { get; set; } = string.Empty;
    public Guid   ReferenceId { get; set; }
    public DateTime SubmittedAt { get; set; }
}
